staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -628,14 +633,17 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 			deleteObj = obj
 			return nil, nil, errEmptiedFinalizers
 		}
+
+		// always return a nil TTL when we have no TTL func so that the
+		// storage interface can distinguish between no TTL, 0 TTL, 1+ TTL
+		if e.TTLFunc == nil {
```

Comment:
Can we eliminate the copy paste of the `e.TTLFunc == nil` checks?  Maybe default TTL function that we use and does the right thing so that we don't have to special case `e.TTLFunc == nil` plus modify `calculateTTL` to return a pointer for TTL?

Reply:
I will see what I can do, the semantics for TTL on create and update are different since create does not care about `nil` whereas update does.